specification/under_development/profiles/Base_Profile.md

```diff
@@ -98,165 +142,349 @@ declare void @__quantum__qis__h__body(%Qubit*)
+| `ret`                    | Used to return the exit code of the program.                                                                     | Must occur (only) as the final instruction of the second (and last) block in the entry point.               |
+| `inttoptr`               | Used to cast an `i64` integer value to either a `%Qubit*` or a `%Result*`.                                       | May be used as part of a function call only.                                                                |
+| `getelementptr inbounds` | Used to create an `i8*` to pass a constant string for the purpose of labeling an output value.                   | May be used as part of a call to an output recording function only.                                           |
 
 ## Data Types and Values
 
-Within the base profile, defining local variables is not supported. Arguments to
-calls correspondingly are expected to be constant values.
-
-Qubits and results are passes as a pointer of type `%Qubit*` and `%Result*`
-respectively, where the pointer itself rather than the memory location
-identifies the qubit or result: For the purpose of passing them as arguments in
-function calls, a 64-bit signed integer value is cast to the appropriate pointer
-type. The signed integer that is case must be in the interval `[0, nrQubits)`
-for `%Qubit*` and `[0, nrResults)` for `%Result*`, where `nrQubits` and
-`nrResults` are the required number of qubits and results defined by entry point
-attributes.
-
-Only the `%Qubit` and `%Result` data types are required to be supported by all
-backends.
-
-Integers and double-precision floating point numbers are available as in full
-QIR; however, computations using these numeric types are not available.
+Within the Base Profile, defining local variables is not supported; instructions
+cannot be nested and constant expressions are fully evaluated. This implies the
+following:
+
+- Call arguments must be constant values, and `inttoptr` casts as well as
+  `getelementptr` instructions must be inlined into a call instruction.
+- It is not possible to express classical computations, such as, e.g., adding
+  two double values, as part of a Base Profile compliant program.
+
+Constants of any type are permitted as part of a function call. What data types
+occur in the program hence depends entirely on the used QIS functions, as well
+as the runtime functions used for output recording. Constant values of type
+`i64` in particular may be used as part of calls to output recording functions;
+see the section on [output recording](#output-recording) for more details.
+
+For a program to be considered valid, it must make use of at least one qubit and
+perform at least one measurement. The `%Qubit*` and `%Result*` data types hence
+must be supported by all backends. Qubits and results can occur only as
```

Comment:
Why is this a requirement, it seems to be babysitting the user to a degree? I can conceive of scenarios where I am trying to test aspects of our stack where I would like to run NOP programs.

Reply:
I know certain backends do not necessarily want to deal with empty programs, but there were some other comments regarding this topic as well. Seeing the general sentiment, I'll remove this requirement from the spec so that a NoOp program is valid.